### PR TITLE
fix: made required changes

### PR DESCRIPTION
**Pull request by NikkyXO**

### CODE REVIEW

**Comment by NikkyXO** [comment]
i have made required changes to the end point  and here is  the required screen shot  of the endpoint testings

The screenshots are as follows
-> screenshots  were made to show file lists being returned  while retreiving the message list and returning retrieving empty files , because no files where uploaded  because uploading files through the Upload file endpoint was unavailable , and then it tried using the  send messages endpoint to pass in a file but it kept giving 422 error which trying to debug was time taking, so i used a dummy urlpath to files on local machine  for testing especially the endpoint helper zipfile file function
screenshot was done while trying to retrieve and view archive files 

the screenshots are attached below, thanks for your kind patience
ps: the previous PR i made was unintentionallly close while trying to pushto my repo because i did a git push force


CheckingArchiveFile
dummyurlpath
ntent.com/96661213/204083479-3b5a4820-1767-435d-965d-1683a8e6eb05.png)

extractArchive
rcontent.com/96661213/204083474-dc23a61b-e8c2-4f38-bc95-0ce971292f8e.png)
ent.com/96661213/204083471-000b292d-c231-4854-9b3e-218da0c19420.png)
ubusercontent.com/96661213/204083469-9e27d904-d3d2-4634-945e-c7a1392ed693.png)

File: backend/endpoints/files.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        fdir, fname = os.path.split(filepath)
+        zip_path = os.path.join(zip_subdir, fname)
+        temp.write(fdir, zip_path)
+
+    temp.close()
+    resp = Response(data.getvalue(), media_type="image/png-compressed", headers={
+        'Content-Disposition': f'attachment;filename={zip_filename}'
+    })
+
+    return response
```
Comment (mikengr) [inline backend/endpoints/files.py]: how did you manage to test this since `response` is not defined anywhere?

File: backend/endpoints/files.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        # room_id: 123456
+# }
+
+
+def zipfiles(filenames: str):
+    zip_subdir = "dummy_archive_path"
+    zip_filename = "archive.zip"
+    """
+        filenames = ["dummy_url path1", "dummy_url path12" ]
+    """
```
Comment (mikengr) [inline backend/endpoints/files.py]: this docstring is wrongly placed, should be after the function definition and properly writing to depict what the function is doing

File: backend/endpoints/files.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+import os
+import zipfile
+import io
+from zipfile import ZipFile
+router = APIRouter()
+
+"""
+ A get method endpoint to Handle retrieving and Viewing of uploaded files in a DM
+
+"""
```
Comment (mikengr) [inline backend/endpoints/files.py]: what is this for? did you miss some codes?

File: backend/endpoints/files.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        # room_id:
+        # 6373eb4f4746182adae97316
+
+        #  Use:
+        # org_id: 619ba4
+        # room_id: 123456
+# }
+
+
+def zipfiles(filenames: str):
```
Comment (mikengr) [inline backend/endpoints/files.py]: this function should probably be in utils dir or file storage dir

File: backend/endpoints/files.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+params:
+    org_id = Organisation id of user
+    room_id = room id of user
+
+Retrieves All Messages in the Dm with the get_room_message method
+Handles  conditons for single and Multiple Uploaded Files
+"""
+
+@router.get("/org/{org_id}/rooms/{room_id}/files")
```
Comment (mikengr) [inline backend/endpoints/files.py:11]: check other endpoints on the codebase to get inspiration on routes definition and endpoint documentation

**Review by mikengr — CHANGES_REQUESTED** [review]
Good work @NikkyXO 
I left some comments

File: backend/endpoints/files.py (hunk 1 above)
Comment (NikkyXO) [reply backend/endpoints/files.py]: return response was a mistake while trying to clean up code to improve readability.. now rectified
endpoint retested and works properly like in my screenshots shared in previous comments

File: backend/endpoints/files.py (hunk 2 above)
Comment (NikkyXO) [reply backend/endpoints/files.py]: i have also corrected placement of the docstrings; placed them inside the function

File: backend/endpoints/files.py (hunk 3 above)
Comment (NikkyXO) [reply backend/endpoints/files.py]: no

File: backend/endpoints/files.py (hunk 4 above)
Comment (NikkyXO) [reply backend/endpoints/files.py]: Done

File: backend/endpoints/files.py (hunk 5 above)
Comment (NikkyXO) [reply backend/endpoints/files.py:11]: i checked other endpoints on the codebase  and saw that it followed this endpoint route patern : "/org/{org_id}/rooms/{room_id}/files"  as this was also suggested to be used by a previous second reviewer

i also referred to other endpoint routes pattern eg; the get message endpoint route pattern "/org/{org_id}/rooms/{room_id}/messages"

**Comment by mikengr** [comment]
I can't even run your code on my local. The server is not responding on this branch.

File: backend/endpoints/files.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+			status_code=status.HTTP_404_NOT_FOUND,
+			detail={"No file was uploaded": room_messages},
+		)
+
+	list_files = []
+	for message in room_messages:
+		if len(message['files']):	
+			list_files.append(message['files'])
+
+	filepaths = [" ".join(item) for item in list_files]
```
Comment (mikengr) [inline backend/endpoints/files.py]: you can rewrite lines 35 to 40 like this
```suggestion
file_paths = []
for message in room_messages:
    if len(message['files']):
         file_paths = file_paths + message['files']
```

File: backend/utils/files_utils.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
+import os
+import zipfile
+import io
+import wget
+from starlette.responses import StreamingResponse, Response
+
+
+
+def zipfiles(filenames: str):
```
Comment (mikengr) [inline backend/utils/files_utils.py]: this is supposed to be a `list` not `str`
Also `filenames` is misleading, perhaps you should use `file_paths`
And improve your variable naming to actually represent the resource it holds. Also stick to one naming convention, it's either you are using snake casing or camel casing not both.

**Review by mikengr — CHANGES_REQUESTED** [review]
@NikkyXO very good work.
I left some quick suggestions. Please fix these quickly so that we can move on.

**Review by mikengr — APPROVED** [review]
LGTM. Very good work @NikkyXO

File: backend/endpoints/files.py (hunk 6 above)
Comment (mikengr) [reply backend/endpoints/files.py]: what happened to this file? you deleted it? @NikkyXO

Comment (NikkyXO) [reply backend/endpoints/files.py]: just saw that now., probabably when deleting the media files.i have restored it now and repushed it

**Pull request merged**